index.js

```diff
@@ -2,48 +2,47 @@
 const mimicFn = require('mimic-fn');
 
 const pTime = fn => {
-	const ret = function () {
+	const wrappedFunction = function (...arguments_) {
 		const start = Date.now();
-		// TODO: use rest/spread when Node.js 6 is targeted
-		const promise = fn.apply(null, arguments);
+		const promise = fn(...arguments_);
 
-		const retPromise = promise.then(res => {
+		const retPromise = promise.then(result => {
 			retPromise.time = Date.now() - start;
-			return res;
-		}, err => {
+			return result;
+		}, error => {
 			retPromise.time = Date.now() - start;
-			throw err;
+			throw error;
 		});
 
 		return retPromise;
 	};
 
-	mimicFn(ret, fn);
+	mimicFn(wrappedFunction, fn);
 
-	return ret;
+	return wrappedFunction;
 };
 
 const log = (fn, promise) => {
 	console.log(`Promise from ${fn.displayName || fn.name || '[anonymous]'} resolved in ${promise.time} ms`);
 };
 
 module.exports = pTime;
+module.exports.default = pTime;
 
 module.exports.log = fn => {
 	const wrapper = pTime(fn);
 
-	return function () {
-		// TODO: use rest/spread when Node.js 6 is targeted
-		const promise = wrapper.apply(null, arguments);
+	return function (...arguments_) {
+		const promise = wrapper(...arguments_);
 
 		promise
-			.then(res => {
+			.then(result => {
```

Comment:
Can you convert to use async/await and require Node.js 8? Applies to all modules really. Except for when the module is already targeting Node.js 6 and wouldn't be able to take advantage of async/await, as that wouldn't be worth doing a breaking change over.

Reply:
OK

Reply:
I thought that I should leave modules on v6 if async-await is not strictly required. Which I thought is not really required here.

Reply:
Sorry for being unclear. I prefer upgrading to Node.js 8 in all cases, except if the last published version already targets Node.js 6 and there's no promise usage in the module, otherwise Node.js 8 and async/await.